geopandas/_compat.py

```diff
@@ -19,6 +19,7 @@
 PANDAS_GE_11 = str(pd.__version__) >= LooseVersion("1.1.0")
 PANDAS_GE_115 = str(pd.__version__) >= LooseVersion("1.1.5")
 PANDAS_GE_12 = str(pd.__version__) >= LooseVersion("1.2.0")
+PANDAS_GE_20 = str(pd.__version__) > LooseVersion("1.4.0.dev")
```

Comment:
```suggestion
PANDAS_GE_14 = str(pd.__version__) >= LooseVersion("1.4.0")
```

The upcoming release of pandas is 1.4 (and it's in this version that the index can now store extension arrays)